Introduction to Tabular data

In [9]:
# Importing the libraries
# OS library is used to interact with os file system. It helps in creating new directory and save visualizations that we have created with the help of python.
import os

# pandas -> used to read and interpret csv file
import pandas as pd

# numpy -> to convert data from csv to array or other required formats
import numpy as np

# seaborn -> uses Matplotlib to plot graphs from the datas. Used to visualize random distributions.
import seaborn as sns

# matplotlib -> to visualize and plot data
import matplotlib.pyplot as plt

In [10]:
# pd.read_csv('https://drive.google.com/uc?id=1BQVc6MHjQFtDC9iP1isT_K4ojVe_Oil-')

# pd.read_csv('/Users/shubhamk/Documents/GitHub/Astronomical_Visualization/cleaned_star_data.csv')

#Storing data in variable star_df
star_df = pd.read_csv('https://raw.githubusercontent.com/Shubham91999/Astronomical_Visualization/main/cleaned_star_data.csv')

#To check the datatype of returned result
type(pd.read_csv('https://raw.githubusercontent.com/Shubham91999/Astronomical_Visualization/main/cleaned_star_data.csv'))

#head() is used to return first 5 rows of the result set
star_df.head()

#sample() is used to analyze the random rows from dataset, here 10 rows are selected randomly
star_df.sample(10)

#Checking general information about the dataframe
star_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 240 entries, 0 to 239
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Temperature (K)         240 non-null    int64  
 1   Luminosity(L/Lo)        240 non-null    float64
 2   Radius(R/Ro)            240 non-null    float64
 3   Absolute magnitude(Mv)  240 non-null    float64
 4   Star type               240 non-null    int64  
 5   Star color              240 non-null    object 
 6   Spectral Class          240 non-null    object 
dtypes: float64(3), int64(2), object(2)
memory usage: 13.3+ KB


In [11]:
star_df.sample(10)


Temperature (K)  Luminosity(L/Lo)  Radius(R/Ro)  Absolute magnitude(Mv)  \
59              3535     195000.000000     1546.0000                  -11.36   
153            16390       1278.000000        5.6800                   -3.32   
174             3600     240000.000000     1190.0000                   -7.89   
201            15680          0.001220        0.0114                   11.92   
95             11250        672.000000        6.9800                   -2.30   
34              5800          0.810000        0.9000                    5.05   
189             3523          0.000957        0.1290                   16.35   
79              3158          0.001350        0.1610                   13.98   
53              3749     550000.000000     1648.0000                   -8.05   
84             14100          0.000670        0.0089                   12.17   

     Star type    Star color Spectral Class  
59           5           Red              M  
153          3    Blue-White              B  
174          5           Red              M  
201          2          Blue              B  
95           3    Blue-White              A  
34           3  Yellow-White              F  
189          0           Red              M  
79           1           Red              M  
53           5           Red              M  
84           2    Blue-White              B

# Obervation

From above information we can say that, the dataset has 240 rows, 6 features and 1 target column.

* Features:
1. Absolute Temperature (in K)
2. Relative Luminosity (L/Lo)
3. Relative Radius (R/Ro)
4. Absolute Magnitude (Mv)
5. Star Color
6. Spectral Class

* Target Column:
Star Type 

Here,
1. Lo = 3.828 x 10^26 Watts (Avg Luminosity of Sun)
2. Ro = 6.9551 x 10^8 m (Avg Radius of Sun)

* Two Categorical features which will need some kind of encoding
1. Star Color
2. Spectral Class

* Information on the star type is given below:
 0 -> Brown Dwarf




